styx-api-service/src/main/java/com/spotify/styx/api/Middlewares.java

```diff
@@ -37,6 +44,32 @@
  */
 public final class Middlewares {
 
+  private static class ClientValidator implements
+                               Middleware<AsyncHandler<? extends Response<?>>,
+                                   AsyncHandler<? extends Response<ByteString>>> {
+    private final Supplier<List<String>> clientBlacklist;
+
+    ClientValidator(Storage storage) {
+      clientBlacklist = new CachedSupplier<>(storage::clientBlacklist, Instant::now);
```

Comment:
Good point. I like this better. Didn't really like the class anyway. We can get rid of it and use closure instead.